admin/used_flusher.go

```diff
@@ -20,6 +22,18 @@ type usedFlusher struct {
 	cancel      context.CancelFunc
 }
 
+func (u *usedFlusher) Deployment(id string) {
+	u.lock.Lock()
+	defer u.lock.Unlock()
+
+	u.deployments[id] = true
+}
+
+func (u *usedFlusher) Close() {
+	u.flush(context.Background())
+	u.cancel()
+}
```

Comment:
There's a raise condition where if `flush` is called in `runBackground` and takes a long time to flush, and then `Close` is called while the first `flush` is running, the second `flush` here will exit quickly (since there's nothing to flush), and then `Close` will return even though the first `flush` is still running (which may lead to process exit and the first flush being stopped).